src/videojs.ads.js

```diff
@@ -373,75 +270,68 @@ var
     // plugins that wish to handle ad events to do so while
     // avoiding the complexity for common usage
     (function() {
-      var
-        videoEvents = videojs.getComponent('Html5').Events,
-        i,
-        returnTrue = function() { return true; },
-        triggerEvent = function(type, event) {
-          // pretend we called stopImmediatePropagation because we want the native
-          // element events to continue propagating
-          event.isImmediatePropagationStopped = returnTrue;
-          event.cancelBubble = true;
-          event.isPropagationStopped = returnTrue;
-          player.trigger({
-            type: type + event.type,
-            state: player.ads.state,
-            originalEvent: event
-          });
-        },
-        redispatch = function(event) {
-          if (player.ads.state === 'ad-playback') {
-            triggerEvent('ad', event);
-          } else if (player.ads.state === 'content-playback' && event.type === 'ended') {
-            triggerEvent('content', event);
-          } else if (player.ads.state === 'content-resuming') {
-            if (player.ads.snapshot) {
-              // the video element was recycled for ad playback
-              if (player.currentSrc() !== player.ads.snapshot.currentSrc) {
-                if (event.type === 'loadstart') {
-                  return;
-                }
-                return triggerEvent('content', event);
-
-              // we ended playing postrolls and the video itself
-              // the content src is back in place
-              } else if (player.ads.snapshot.ended) {
-                if ((event.type === 'pause' ||
-                    event.type === 'ended')) {
-                  // after loading a video, the natural state is to not be started
-                  // in this case, it actually has, so, we do it manually
-                  player.addClass('vjs-has-started');
-                  // let `pause` and `ended` events through, naturally
-                  return;
-                }
-                // prefix all other events in content-resuming with `content`
-                return triggerEvent('content', event);
+      var videoEvents = VIDEO_EVENTS.concat([
+        'firstplay',
+        'loadedalldata'
+      ]);
+
+      var returnTrue = function() { return true; };
+
+      var triggerEvent = function(type, event) {
+        // pretend we called stopImmediatePropagation because we want the native
+        // element events to continue propagating
+        event.isImmediatePropagationStopped = returnTrue;
+        event.cancelBubble = true;
+        event.isPropagationStopped = returnTrue;
+        player.trigger({
+          type: type + event.type,
+          state: player.ads.state,
+          originalEvent: event
+        });
+      };
+
+      player.on(videoEvents, function redispatch(event) {
```

Comment:
I assume the redispatch function is the same just moved? The diff makes it a bit hard to see it fully/exactly.